src/server/game/Handlers/GroupHandler.cpp

```diff
@@ -742,6 +743,17 @@ void WorldSession::HandleRaidReadyCheckOpcode(WorldPacket& recvData)
             return;
         /********************/
 
+        // Check if Ready Check in BG is enabled
+        if (sWorld->getBoolConfig(CONFIG_BATTLEGROUND_DISABLE_READY_CHECK_IN_BG))
+        {
+            // Check if player is in BG and trying to send Ready Check
+            if (group->IsLeader(GetPlayer()->GetGUID()) && _player->InBattleground())
```

Comment:
On Line 741, there is already a leader check or do you explicit need to check not for Asisstant?

Reply:
Hm, didn't think of that, I took the code that existed in the issue and added it within the config.
Makes sense what you say though, assistants could theoretically use it still

I could test with only _player->InBattleground();